CaImAn/caiman_main.py

```diff
@@ -0,0 +1,316 @@
+# Import miscellaneous and utilities librarys
+import os
+import sys
+import h5py
+import cv2
+import numpy as np
+import tempfile
+from tifffile import imwrite
+import logging
+from typing import Any, Dict, List, Optional, Tuple, Union, Callable
+
+sys.path.append("..")
+# Import CaimAn related utilities libraries
+import utilities as utils
+import definitions as defs
+import caiman_definitions as cm_defs
+import caiman_parameters  as cm_params
+
+# Import for PyInstaller
+from multiprocessing import freeze_support
+freeze_support()
+
+
+def main(caiman_parameters):
+    """
+    CaImAn CNMF algorithm
+    """
+
+        # Import for CaimAn lib
+    import caiman as cm
+    from caiman.cluster import setup_cluster
+    from caiman.source_extraction import cnmf
+    from caiman.source_extraction.cnmf import params
+    from caiman.motion_correction import MotionCorrect
+
+    #os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
+
+    try:
+        cv2.setNumThreads(0)
+    except:
+        pass
+
+    # start the cluster
+    try:
+        cm.stop_server()  # stop it if it was running
+    except():
+        pass
+
+    c, dview, n_processes = setup_cluster(backend="local", n_processes=None, single_thread=False)
+
+    file_ = h5py.File(caiman_parameters.paths[defs.Parameters.Path.FILEPATH], 'r')
+    if defs.DoricFile.Dataset.IMAGE_STACK in file_[caiman_parameters.paths[defs.Parameters.Path.H5PATH]]:
+        images = np.array(file_[f"{caiman_parameters.paths[defs.Parameters.Path.H5PATH]}/{defs.DoricFile.Dataset.IMAGE_STACK}"])
```

Comment:
set dataname, then load data

Reply:
` images = np.array(file_[f"{caiman_parameters.paths[defs.Parameters.Path.H5PATH]}/dataname"])`

Reply:
and the if condition for dataname setting can be written in one line